### PR TITLE
[Keyboard, Web] Map from "Esc" to the Escape key

### LINKED ISSUE #106062: Mappings for 'Esc' KeyboardEvent key values are missing.
KeyboardEvent key values in keyboard_maps.dart have mappings for 'Escape' but not for the older 'Esc' key values.

These values were used by older IE and Firefox browsers, and are still being generated by the Cobalt browser. 

(Specifically, this is causing problems when running a Flutter-Web app on Cobalt and using the back button on the TV remote.  The KeyboardEvent generated by Cobalt has an 'Esc' key value, which then does not get interpreted by Flutter as an LocalKeyboardKey.escape event.)

See the note on the 'Escape' value here:
https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values#ui_keys

### PR DESCRIPTION
**Pull request by dkwingsmt**
This PR fixes https://github.com/flutter/flutter/issues/106062 and adds the following entries to the Web key mapping:
- Code `"Esc"` to physical key `esc`.
- Key `"Esc"` to logical key `esc`.

The change to `PhysicalKeyEntry` is a bit tricky, since it only allows one `chromeName`. I didn't want to change `chromeName` to a list of string and cause hundreds of lines of changes in `physical_key_data.json`, so I added a new field `otherWebNames`.

The test is co-authored with @egramond from https://github.com/flutter/flutter/commit/58fa5754dc31dab90f0068615654a409f9be8fe2#diff-ac9c378fae8af8f4ffb68fad0b56e14f9f15ab797a0e7271a3690317a4df9c59.

All mapping changes will also be applied to the engine in a separate PR.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by egramond** [comment]
Looks great. Thank you!

File: dev/tools/gen_keycodes/data/logical_key_data.json [hunk 1]
```diff
@@ -1552,7 +1552,8 @@
     "value": 4294967323,
     "names": {
       "web": [
```
Comment (Sdxr888) [inline dev/tools/gen_keycodes/data/logical_key_data.json:1554]: https://www.facebook.com/profile.php?id=100053958475496

File: dev/tools/gen_keycodes/data/supplemental_hid_codes.inc [hunk 2]
```diff
@@ -50,6 +50,11 @@
   DOM_CODE(0x05ff1e, 0x0000, 0x0000, 0x0000, 0xffff, "GameButtonY", BUTTON_Y),
   DOM_CODE(0x05ff1f, 0x0000, 0x0000, 0x0000, 0xffff, "GameButtonZ", BUTTON_Z),
 
+  // Sometimes the Escape key produces "Esc" instead of "Escape". This includes
+  // older IE and Firefox browsers, and the current Cobalt browser.
+  // See: https://github.com/flutter/flutter/issues/106062
```
Comment (Sdxr888) [inline dev/tools/gen_keycodes/data/supplemental_hid_codes.inc:55]: https://www.facebook.com/profile.php?id=100053958475496

**Pull request merged**